tools/src/main/java/org/apache/kafka/tools/streams/StreamsGroupCommand.java

```diff
@@ -330,13 +403,433 @@ Map<TopicPartition, OffsetsInfo> getOffsets(StreamsGroupDescription description)
+                Map<String, KafkaFuture<StreamsGroupDescription>> streamsGroups = adminClient.describeStreamsGroups(
+                    groupIds
+                ).describedGroups();
+
+                streamsGroups.forEach((groupId, groupDescription) -> {
+                    try {
+                        String state = groupDescription.get().groupState().toString();
+                        switch (state) {
+                            case "Empty":
+                            case "Dead":
+                                result.put(groupId, resetOffsetsForInactiveGroup(groupId));
+                                break;
+                            default:
+                                printError("Assignments can only be reset if the group '" + groupId + "' is inactive, but the current state is " + state + ".", Optional.empty());
+                                result.put(groupId, Collections.emptyMap());
+                        }
+                    } catch (InterruptedException ie) {
+                        throw new RuntimeException(ie);
+                    } catch (ExecutionException ee) {
+                        if (ee.getCause() instanceof GroupIdNotFoundException) {
+                            result.put(groupId, resetOffsetsForInactiveGroup(groupId));
+                        } else {
+                            throw new RuntimeException(ee);
+                        }
+                    }
+                });
+            }
+            return result;
+        }
+
+        private Map<TopicPartition, OffsetAndMetadata> resetOffsetsForInactiveGroup(String groupId) {
+            try {
+                Collection<TopicPartition> partitionsToReset = getPartitionsToReset(groupId);
+                Map<TopicPartition, OffsetAndMetadata> preparedOffsets = prepareOffsetsToReset(groupId, partitionsToReset);
+
+                // Dry-run is the default behavior if --execute is not specified
+                boolean dryRun = opts.options.has(opts.dryRunOpt) || !opts.options.has(opts.executeOpt);
+                if (!dryRun) {
+                    adminClient.alterStreamsGroupOffsets(
+                        groupId,
+                        preparedOffsets
+                    ).all().get();
+                }
+
+                return preparedOffsets;
+            } catch (InterruptedException ie) {
+                throw new RuntimeException(ie);
+            } catch (ExecutionException ee) {
+                Throwable cause = ee.getCause();
+                if (cause instanceof KafkaException) {
+                    throw (KafkaException) cause;
+                } else {
+                    throw new RuntimeException(cause);
+                }
+            }
+        }
+
+        private Collection<TopicPartition> getPartitionsToReset(String groupId) throws ExecutionException, InterruptedException {
+            if (opts.options.has(opts.allTopicsOpt)) {
+                return getCommittedOffsets(groupId).keySet();
```

Comment:
I think this may not be the correct behavior. We only want to reset the offset of source topics, never the offset of repartition topics (which are also committed). For a clean reset, as implemented by the streams resetter, I think we want to delete the internal topics (through the other options). Resetting repartition topics to the beginning seems incorrect, since it will mean that we will reconsume these records in the downstream subtopologies.

@bbejeck Wdyt?

We are already describing the streams group, so I think this shouldn't be too hard to implement - we have to pass the source topics into this function.